pkg/testutil/random.go

```diff
@@ -65,3 +65,14 @@ func NewRandomReader(rand *rand.Rand, size int64) io.Reader {
 func UniqueName() string {
 	return nanoid.MustGenerate(chars, charsSize)
 }
+
+func RandomS3Path(rand *rand.Rand, size int) string {
+	// List of valid chars per https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
+	var pathRunes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!-_.*'()&$@=;/:+,?")
+
+	b := make([]rune, size)
+	for i := range b {
+		b[i] = pathRunes[rand.Intn(len(pathRunes))]
+	}
+	return string(b)
+}
```

Comment:
- Can we make the charset constant?
- Can we use `gonanoid.Generate` to generate the random key